// @ionic/core/dist/cjs/index.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const config = require('./config-773d6543.js');
const animation = require('./animation-77f06c37.js');
const index = require('./index-54b638ff.js');
const constants = require('./constants-e9c8ab68.js');
const overlays = require('./overlays-16c49da8.js');

const setupConfig = (config) => {
    const win = window;
    const Ionic = win.Ionic;
    if (Ionic && Ionic.config && Ionic.config.constructor.name !== 'Object') {
        console.error('ionic config was already initialized');
        return;
    }
    win.Ionic = win.Ionic || {};
    win.Ionic.config = Object.assign({}, win.Ionic.config, config);
    return win.Ionic.config;
};

exports.getPlatforms = config.getPlatforms;
exports.isPlatform = config.isPlatform;
exports.createAnimation = animation.createAnimation;
exports.createGesture = index.createGesture;
exports.LIFECYCLE_DID_ENTER = constants.LIFECYCLE_DID_ENTER;
exports.LIFECYCLE_DID_LEAVE = constants.LIFECYCLE_DID_LEAVE;
exports.LIFECYCLE_WILL_ENTER = constants.LIFECYCLE_WILL_ENTER;
exports.LIFECYCLE_WILL_LEAVE = constants.LIFECYCLE_WILL_LEAVE;
exports.LIFECYCLE_WILL_UNLOAD = constants.LIFECYCLE_WILL_UNLOAD;
exports.actionSheetController = overlays.actionSheetController;
exports.alertController = overlays.alertController;
exports.loadingController = overlays.loadingController;
exports.modalController = overlays.modalController;
exports.pickerController = overlays.pickerController;
exports.popoverController = overlays.popoverController;
exports.toastController = overlays.toastController;
exports.setupConfig = setupConfig;
